File: Ruby/experiment_results.rb
class ExperimentResults


  def initialize(name, mes, avg, med, high, low)
    @@name = name
    @@mes = mes
    @@avg = avg
    @@med = med
    @@high = high
    @@low = low
  end



  def PrintSelf()

    puts @@name+" Results: "
    puts "Median: "+@@med.round.to_s+"ns"
    puts "Average: "+@@avg.round.to_s+"ns"
    puts "High: "+@@high.round.to_s+"ns"
    puts "Low: " + @@low.round.to_s+ "ns"
    puts ""
    puts "===================="
    puts ""
  end

  def name
    @@name
  end

  def mes
    @@mes
  end

  def avg
    @@avg
  end

  def med
    @@med
  end

  def high
    @@high
  end

  def low
    @@low
  end


end

File: Rust/src/ExperimentResults.rs


pub mod results{

    pub struct experimentResults{
        pub average: f32,
        pub median: u64,
        pub high: u64,
        pub low: u64
    }

    impl experimentResults {

        pub fn PrintMe(&self){
            println!("");
            println!("Median: {}ns", self.median);
            println!("Average: {}ns", self.average);
            println!("High: {}ns", self.high);
            println!("Low: {}ns", self.low);
            println!("===========================")
        }



    }


}